src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts

```diff
@@ -112,7 +112,7 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
             return Promise.resolve({ ...payment, paymentData: this._formattedPayload(nonce) });
         }
 
-        if (isVaultedInstrument(paymentData)) {
+        if (isVaultedInstrument(paymentData) || isHostedVaultedInstrument(paymentData)) {
```

Comment:
Good point. I'll double check.

Reply:
It turns out if we don't have this check, TS thinks `paymentData` could either be `HostedVaultedInstrument` or `HostedInstrument`. Below this line, we're trying to read `shouldSaveInstrument` from `paymentData`. But `shouldSaveInstrument` doesn't exist for `HostedVaultedInstrument`. Therefore TS will throw an error without this check.

Reply:
I think the issue is that `pick` that we are doing. Do you think it will work if we remove the properties from the base type and have a new type for CreditCard? Something on these lines:

```ts
export type HostedVaultedInstrument = VaultedInstrument;

export interface VaultedInstrument {
    instrumentId: string;
}

export interface CardVaultedInstrument extends VaultedInstrument {
    ccCvv?: string;
    ccNumber?: string;
}

export interface ThreeDSVaultedInstrument extends CardVaultedInstrument {
    iin?: string;
    threeDSecure?: ThreeDSecure | ThreeDSecureToken;
}
```

Reply:
Thanks for the suggestion. I gave it a go but unfortunately it didn't help. 😢